src/main/java/org/myrobotlab/service/InMoov.java

```diff
@@ -154,9 +161,39 @@ public InMoov(String n) {
   // ---------------------------------------------------------------
 
   public void attach(Attachable attachable) {
+    //opencv
     if (attachable instanceof OpenCV) {
       opencv = (OpenCV) attachable;
+      subscribe(opencv.getName(), "publishClassification");
+      //SpeechRecognizer
+    } else if (attachable instanceof SpeechRecognizer) {
+      if (attachable.getClass().getSimpleName().equals("WebkitSpeechRecognition")) {
+        WebGui webgui = (WebGui) Runtime.create("webgui", "WebGui");
+        webgui.autoStartBrowser(false);
+        webgui.startService();
+        webgui.startBrowser("http://localhost:8888/#/service/" + attachable.getName());
+      }
+      ear = (SpeechRecognizer) attachable;
+    } else if (attachable instanceof ProgramAB) {
+      if (htmlFilter == null) {
+        htmlFilter = (HtmlFilter) Runtime.start(this.getIntanceName() + ".htmlFilter", "HtmlFilter");
+      }
+      chatBot = (ProgramAB) attachable;
+    }
+    //sub attach
+    if (chatBot != null && ear != null) {
+      chatBot.attach((Attachable) ear);
```

Comment:
nope.. all of this stuff should be in python.. and not here.  Do not handle all of this logic here.  It's way too much of a change before the nixie release.  Changes of this scale are for Ogre at this point and NOT for Nixie.

Reply:
ok let's do it Inside ogre ...